### PR TITLE
implement lib

### PR DESCRIPTION
**Pull request by msokoloff1**
* Use library for shared etl functions.

### CODE REVIEW

File: src/jobs/etl/images/bounding_box/bounding_box_etl.py [hunk 1]
```diff
@@ -77,32 +71,31 @@ def process_label(label: Label, bucket: Bucket) -> str:
...
-                "yMax": round(bbox.end.y / h, 2),
+                **convert_to_percent(bbox, w, h)
             })
 
     if len(bounding_box_annotations) == 0:
-        return
+        raise InvalidLabelException(f"There are 0 valid annotations for data row `{label.data.uid}`.")
 
+    gcs_uri = upload_image_to_gcs(image_bytes, label.data.uid, bucket, (w, h))
+    del image_bytes
```
Comment (gszpak) [inline src/jobs/etl/images/bounding_box/bounding_box_etl.py:91]: I think you need gc.collect() for `del` to make sense

File: src/training_lib/setup.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import setuptools
+
+setuptools.setup(
+    name="training_lib",
+    version="0.0.1",
+    author="Labelbox",
+    author_email="engineering@labelbox.com",
+    description="shared code for ",
```
Comment (gszpak) [inline src/training_lib/setup.py:8]: for?:)

File: src/training_lib/setup.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    packages=setuptools.find_packages(),
+    install_requires=["PILLOW"],
```
Comment (gszpak) [inline src/training_lib/setup.py:10]: gcloud, labelbox[data]?

File: src/training_lib/training_lib/storage.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            The new dimensions will be (width * 1/downsample_factor, height * 1/downsample_factor)
+    Returns:
+        The resized PIL Image
+    """
+    try:
+        with _download_image(image_url) as image:
+            w,h = image.size
+            with image.resize((int(w *  1./downsample_factor), int(h * 1./downsample_factor))) as resized_image:
+                image_bytes = image_to_bytes(resized_image)
+                resized_image.close()
```
Comment (gszpak) [inline src/training_lib/training_lib/storage.py:58]: no need to close context manager

**Review by gszpak — APPROVED** [review]
lgtm, mostly nits, although it wasn't my most thorough review ever:)

File: src/jobs/etl/images/bounding_box/bounding_box_etl.py (hunk 1 above)
Comment (msokoloff1) [reply src/jobs/etl/images/bounding_box/bounding_box_etl.py:91]: oops there was a mem leak initially and this was left from my experimentation...

File: src/training_lib/setup.py (hunk 2 above)
Comment (msokoloff1) [reply src/training_lib/setup.py:8]: :(

File: src/training_lib/setup.py (hunk 3 above)
Comment (msokoloff1) [reply src/training_lib/setup.py:10]: :(

File: src/training_lib/training_lib/storage.py (hunk 4 above)
Comment (msokoloff1) [reply src/training_lib/training_lib/storage.py:58]: :(

**Pull request merged**